File: 594.LongestHarmoniousSubsequence/subsequence.cc
class Solution {
public:
    int findLHS(vector<int>& nums) {
        map<int,int> counts;
        for (int n : nums) {
            counts[n]++;
        }

        int maxLen = 0;
        for (map<int,int>::iterator it = counts.begin(); it != counts.end(); ++it) {
            int key = it->first;
            if (counts.find(key-1) != counts.end()) {
                maxLen = max(maxLen, counts[key] + counts[key-1]);
            }
            if (counts.find(key+1) != counts.end()) {
                maxLen = max(maxLen, counts[key] + counts[key+1]);
            }
        }

        return maxLen;
    }
};

File: 521.LongestUncommonSubsequenceI/Subsequence.java
class Solution {
    public int findLUSlength(String a, String b) {
        if (a.length() != b.length()) {
            return Math.max(a.length(), b.length());
        }

        if (!a.equals(b)) {
            return a.length();
        }

        return -1;
    }
}
